### PR TITLE
Use metafunc generator

### PR DESCRIPTION
**Pull request by amarmulyak**

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -16,7 +16,21 @@ def browser_fixture():
 
 @pytest.fixture
 def data_fixture_js():
-    json_file = open("../test_input_data/qa.json")
+    # json_file = open("../test_input_data/qa.json")
+    json_file = open("C:\\Users\\Andrii\\repositories\\AddressBook\\test_input_data\\qa.json")
```
Comment (velocyno) [inline tests/conftest.py:20]: We should not hardcode the path here
Lets use abspath

File: tests/conftest.py [hunk 2]
```diff
@@ -16,7 +16,21 @@ def browser_fixture():
...
     data_from_file = json.load(json_file)
     yield data_from_file
     json_file.close()
+
+
+# used in generator tests
+def pytest_generate_tests(metafunc):
+    if "data_gen" in metafunc.fixturenames:
+        file = open("C:\\Users\\Andrii\\repositories\\AddressBook"
```
Comment (velocyno) [inline tests/conftest.py:29]: We should not hardcode the path here
Lets use abspath

**Review by velocyno — CHANGES_REQUESTED** [review]

File: tests/test_addresses.py [hunk 3]
```diff
@@ -8,8 +8,11 @@
 
 class TestManageAddresses:
     def test_add_address(self, browser_fixture, data_fixture_js):
-        session_email = data_fixture_js["session_email"]
-        session_password = data_fixture_js["session_password"]
+        # session_email = data_fixture_js["session_email"]
+        # session_password = data_fixture_js["session_password"]
+
+        session_email = "mymail@i.ua"
+        session_password = "123456"
```
Comment (velocyno) [inline tests/test_addresses.py:15]: Why do we use hardcode here?

File: tests/test_addresses.py [hunk 4]
```diff
@@ -142,6 +145,7 @@ def test_add_address(self, browser_fixture, data_fixture_js):
 
     def test_show_address(self, browser_fixture, data_fixture_js):
         addresses = AddressesSearchHelper(browser_fixture)
+        time.sleep(2)
```
Comment (velocyno) [inline tests/test_addresses.py:148]: We use that to wait something?

File: tests/test_negative_edit_address.py [hunk 5]
```diff
@@ -91,11 +91,10 @@ def test_edit_address_negative(
     ):
         if not TestEditAddressNegative.before_all:
             before_all = TestHelper()
-            before_all.create_user(browser_fixture)
+            # before_all.create_user(browser_fixture)
```
Comment (velocyno) [inline tests/test_negative_edit_address.py:94]: why we are using commented code on GitHub? :)

File: tests/setup/setup_negative_test_add_address.py [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+from webdriverdownloader import ChromeDriverDownloader
+
+
+class TestAddAddressNegative:
+
+    # Question: Can I use PARAMETRIZE HERE?
+    @classmethod
+    def setup_class(cls):
+        # file = open("C:\\Users\\Andrii\\repositories\\AddressBook\\test_input_data\\qa.json")
+        cls.file = open("C:\\Users\\Andrii\\repositories\\AddressBook\\test_input_data\\qa.json")
```
Comment (velocyno) [inline tests/setup/setup_negative_test_add_address.py:16]: Oops, we have one more hardcode

File: conftest.py [hunk 7]
```diff
@@ -26,8 +34,9 @@ def data_fixture_js():
 # used in generator tests
 def pytest_generate_tests(metafunc):
     if "data_gen" in metafunc.fixturenames:
-        file = open("C:\\Users\\Andrii\\repositories\\AddressBook"
-                    "\\test_input_data\\qa.json")
+        cur_path = pathlib.Path(__file__).parent
+        # file = open("../test_input_data/qa.json")
```
Comment (velocyno) [inline conftest.py:38]: commented code on GitHub ? :)

File: conftest.py [hunk 8]
```diff
@@ -14,10 +15,17 @@ def browser_fixture():
     driver.quit()
 
 
+# @pytest.fixture
```
Comment (velocyno) [inline conftest.py:18]: oops
commented code again

**Review by velocyno — CHANGES_REQUESTED** [review]

File: tests/test_addresses.py (hunk 3 above)
Comment (amarmulyak) [reply tests/test_addresses.py:15]: These two rows are fixed in "docker" PR

Comment (amarmulyak) [reply tests/test_addresses.py]: Fixed

**Review by velocyno — APPROVED** [review]

**Pull request merged**